### PR TITLE
Add parallelism support for Normal and Beta projection networks.

### PR DESCRIPTION
**Pull request by breakds**
# Motivation

Creating mixture of policy style projection network requires adding an extra `replica` dim to the batch shape of the output distribution, serving as the number of component. This should be done by calling the `make_parallel()` of the corresponding projection network.

# Solution

Added a `parallelism` argument to the affected projection networks for the above purpose. By default the argument is set to `None` so that the behavior of existing code does not change.

The `make_parallel` will now just delegates to this `parallelism` argument. This is slightly better than the naive `make_parallel` as it uses `ParallelFC` under the hood. This is an effortless change.

As some side notes, 

1. `ParallelNormalProjectionNetwork` is now removed in favor of the new argument, which also helps reduce code duplication.
2. Replace a few `as_tensor` with `torch.tensor` to avoid the warnings. Those ocurrences should not affect performance.

# Testing

Added unit tests.

### CODE REVIEW

**Comment by hnyu** [comment]
Did you try directly calling ``net.make_parallel()``? Usually the naive parallel network is good enough.

**Comment by breakds** [comment]
> Did you try directly calling `net.make_parallel()`? Usually the naive parallel network is good enough.

Thanks for pointing this out! Just check it out and it should work. I think the only difference is `ParallelFC` vs for loop of `FC`s. Given the effort is not that big, I would slightly prefer the current solution.

**Comment by Haichao-Zhang** [comment]
> Did you try directly calling `net.make_parallel()`? Usually the naive parallel network is good enough.

Agree with @hnyu 's comment. We can potentially just use ``proj_net.make_parallel()`` to create network replica, serving as the components in the mixture policy.

Another point is that implementations like ```ParallelNormalProjectionNetwork``` follows ALF's conventional on network parallelism, while the proposed form is not.

**Comment by breakds** [comment]
> > Did you try directly calling `net.make_parallel()`? Usually the naive parallel network is good enough.
> 
> Agree with @hnyu 's comment. We can potentially just use `proj_net.make_parallel()` to create network replica, serving as the components in the mixture policy.
> 
> Another point is that implementations like `ParallelNormalProjectionNetwork` follows ALF's conventional on network parallelism, while the proposed form is not.

Agree with that. I added back `make_parallel` so that it is still the universal interface and follows the convention.

`ParallelNormalProjectionNetwork` itself shares quite some implementation details of complex logic with `NormalProjectionNetwork`. I think it is better to merge it with the `NormalProjectionNetwork`. On the other hand, I can add a `ParallelNormalProjectionNetwork` that delegates to `NormalProjectionNetwork` if needed for compatibility.

**Review by emailweixu — APPROVED** [review]

**Review by Haichao-Zhang — APPROVED** [review]
